types/field_type_builder.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+// FieldTypeBuilder constructor
+type FieldTypeBuilder struct {
+	ft FieldType
+}
+
+// NewFieldTypeBuilder will allocate the builder on the stack.
+func NewFieldTypeBuilder() FieldTypeBuilder {
```

Comment:
The motivation is to have a chain builder to replace `newFt := xxx(); newFt.SetXXX(); newFt.SetXXX();`. Of course, `newFT()` could be of type `fn () *FieldType` such that `NewFT().SetXXX().SetXXX()` can be implemented, but that may be escaped while plain `FieldType` is embedding in some structs. If `newFT()` returns a pointer, then it can not be inlined.

Another point is to wrap around `FieldType`, to make the construction consistent. There are too many ways to construct `FieldType`, by exported types or various constructors. I hope we could build `FieldType` by only one API.